spec/support/matchers.rb

```diff
@@ -28,6 +28,33 @@
   end
 end
 
+RSpec::Matchers.define :link_consistently do
+  invalid_links = []
+
+  match do |actual|
+    doc = actual
+
+    good_uri = URI.parse(SITE_URL)
+    http_uri = good_uri.clone.tap { |u| u.scheme = 'http' }
+    www_uri = good_uri.clone.tap { |u| u.host = "www.#{u.host}" }
+    http_www_uri = http_uri.clone.tap { |u| u.host = "www.#{u.host}" }
+
+    invalid_links = doc.css([
+      "a[href^='#{http_uri}']",
+      "a[href^='#{www_uri}']",
+      "a[href^='#{http_www_uri}']",
+      "a[href^='#{good_uri}']:not([href$='/']):not([href$='.pdf'])",
+      "a[href^='/']:not([href$='/']):not([href$='.pdf'])",
+    ].join(',')).map { |node| node.to_html }.to_a
+
+    expect(invalid_links).to be_empty
```

Comment:
would it be easier to iterate over all the `a` tags, and iterate over them in code, instead of relying on CSS/regex syntax?

rough idea:

```ruby
invalid_links = doc.css('a').select do |a|
  uri = URI(a[:href])
  uri.host != 'www.login.gov' ||
    uri.scheme != 'https'
    !uri.path.end_with?('/')
  # etc etc
end
```

Reply:
> would it be easier to iterate over all the `a` tags, and iterate over them in code, instead of relying on CSS/regex syntax?

I like that idea 👍